mypy/reachability.py

```diff
@@ -127,10 +135,15 @@ def consider_sys_version_info(expr: Expression, pyversion: Tuple[int, ...]) -> i
     op = expr.operators[0]
     if op not in ('==', '!=', '<=', '>=', '<', '>'):
         return TRUTH_VALUE_UNKNOWN
+
+    index = contains_sys_version_info(expr.operands[0])
     thing = contains_int_or_tuple_of_ints(expr.operands[1])
-    if thing is None:
+    if index is None or thing is None:
+        index = contains_sys_version_info(expr.operands[1])
+        thing = contains_int_or_tuple_of_ints(expr.operands[0])
+        op = reverse_op[op]
+    if index is None or thing is None:
```

Comment:
This check is redundant, since the conditions below won't trigger if either value is None.

Reply:
That's right. Good point 👍